pysmt/factory.py

```diff
@@ -97,12 +100,60 @@ def get_solver(self, quantified=False, name=None, logic=None):
             closer_logic = get_closer_logic(SolverClass.LOGICS, logic)
             return SolverClass(environment=self.environment,
                                logic=closer_logic,
-                               options=None)
+                               user_options={"generate_models":True})
         else:
             raise NoSolverAvailableError("No solver is available for logic %s" %\
                                          logic)
 
 
+    def get_unsat_core_solver(self, quantified=False, name=None,
```

Comment:
We should discuss this.
In practice we want to select a solver supporting unsat cosera AND configure it for unsat cores...